tasks/nc_installation.yml

```
@@ -175,3 +175,32 @@
     - config
     - themes
     - updater
+
+- name: "[NC] - Disable Nextcloud apps"
+  when: nextcloud_disable_apps is defined
+  block:
+    - name: "[NC apps] - Lists the number of apps available in the instance."
+      ansible.builtin.command: php occ app:list --output=json_pretty --no-warnings
+      args:
+        chdir: "{{ nextcloud_webroot }}"
+      become_user: "{{ nextcloud_websrv_user }}"
+      become: true
+      become_flags: "{{ ansible_become_flags | default(omit) }}"
+      changed_when: false
+      check_mode: false
+      register: nc_apps_list
+
+    - name: "[NC apps] - Convert list to yaml."
+      ansible.builtin.set_fact:
+        nc_available_apps: "{{ nc_apps_list.stdout | from_json }}"
+
+    - name: "[NC apps] - Disable apps"
+      ansible.builtin.command: php occ app:disable "{{ item }}"
+      become_user: "{{ nextcloud_websrv_user }}"
+      become_flags: "{{ ansible_become_flags | default(omit) }}"
+      become: true
+      register: _occ_app_disable_result
+      changed_when:
+        - "'disabled' in _occ_app_disable_result.stdout"
+        - "'No such app' not in _occ_app_disable_result.stdout"
+      with_items: "{{ nextcloud_disable_apps }}"
```

My last suggestion was incomplete.
the block can be dropped and the indentation for the simple task remaining must be adjusted.
I recommand testing this step in CI by disabling some apps in the molecule tests
1 that is never shipped (apporder ?) and 1 that is always shipped (photo ?)

```suggestion
  - name: "[NC apps] - Disable Nextcloud apps"
    ansible.builtin.command: php occ app:disable "{{ item }}"
    become_user: "{{ nextcloud_websrv_user }}"
    become_flags: "{{ ansible_become_flags | default(omit) }}"
    become: true
    register: _occ_app_disable_result
    changed_when:
      - "'disabled' in _occ_app_disable_result.stdout"
      - "'No such app' not in _occ_app_disable_result.stdout"
    with_items: "{{ nextcloud_disable_apps }}"
```